qcodes/dataset/data_set.py

```diff
@@ -1491,8 +1501,35 @@ def _enqueue_results(
 
         if standalones:
             stdln_dict = {st: result_dict[st] for st in standalones}
+            for st in standalones:
+                new_results[st.name] = {
+                    st.name: self._reshape_array_for_cache(st, result_dict[st])
+                }
             self._results += self._finalize_res_dict_standalones(stdln_dict)
 
+        if self._in_memory_cache:
```

Comment:
shall the calls to _reshape_array_for_cache also be done with `if self._in_memory_cache` guard? wouldn't look beautiful, but it is what it is, i guess.

Reply:
I thought about that and my initial assertion was that the ugly code was worse than the performance hit suffered from doing this but happy to change my mind

Reply:
i think that the current ugliness of the code can't get significantly worse ;) however, not calling _reshape_array_for_cache  if _in_memory_cache is False makes the ugly code clearer. otherwise, the way it is now, it is ugly and confusing, as opposed to just ugly. would you agree to this reasoning?

Reply:
Not sure that I agree but lets try it